docs/api/covidcast-signals/dsew-cpr.md

```diff
@@ -36,14 +40,20 @@ County, MSA, state, and HHS-level values are pulled directly from CPR; nation-le
 
 For counts-based fields like hospital admissions, CPR reports rolling sums for the preceding 7 days. The 7-day average signals are computed by Delphi by dividing each sum by 7 and assigning it to the last date in the included range, so e.g. the signal for June 7 is the average of the underlying data for June 1 through 7, inclusive.
 
-The `confirmed_admissions_covid_1d_7dav` signal mirrors the `Confirmed COVID-19 admissions - last 7 days` CPR field for all geographic resolutions except nation. Nation-level admissions is calculated by summing state-level values.
+The `confirmed_admissions_covid_1d_7dav` signal mirrors the `Confirmed COVID-19 admissions - last 7 days` CPR field for all geographic resolutions except nation. Nation-level admissions is calculated by summing state-level values. 
+
+The same holds for `doses_admin_7dav` and `booster doses_admin_7dav` mirroring `Doses administered - last 7 days` and `Booster doses administered - last 7 days`. Nation-level doses are calculated by summing state-level values.
 
 ## Limitations
 
 Nation-level estimates may be inaccurate since aggregations are done using state-level smoothed values instead of raw values. Ideally we would aggregate raw values before smoothing, but the raw values are not accessible in this case.
 
 Because DSEW does not provide updates on weekends, estimates are not available for all dates.
 
+Currently, of all the vaccination signals, county-level data is only available for `people_full_vaccinated`. Until 2021-11-15, there were unallocated vaccinated people reported for different states, which are noted by '000' at the end of the FIPs code in the county level. 
+
+There are many high-visibility data notes from the CPR spreadsheets that explain corrections like decresases in cumulative values for state/HHS/national counts. For example 'On 02/09/2022, SC issued corrections to their vaccination data, resulting in negative values for those initiating vaccination ages 65+.' These can be accessed through the first tab on downloaded CPR worksheets from relevant date ranges (if available)
```

Comment:
```suggestion
This data source is susceptible to large corrections that can create strange data effects such as negative counts and sudden changes of 1M+ counts from one day to the next. Many of these corrections are documented in the "High-Visibility Data Notes" section in the first tab of the CPR spreadsheet for that day. To locate the correct spreadsheet for some `time_value` R, consult the following table:

| Signal type | CPR date |
| - | - |
| Test positivity | R+3, R+10 |
| Test volume | R+7, R+14 |
| Vaccinations | R |

```

Reply:
I'm not sure about R+2, can someone double verify this? Everything else sounds much better :)

Reply:
Keep in mind that lag is not the same as the delta between the reference date and the CPR date. For example, the CPR for 20220218 was uploaded to healthdata.gov on 20220222; the CPR for 20220217 was uploaded on 20220218, etc. So lag _does_ need to account for no uploads on weekends.

For this table, we care only about the time delta between the date in the CPR filename and the reference date in the header for a particular signal type.

I've exhaustively checked and found ... not what I expected. Here's the table of deltas for hospital admissions. Each cell represents the fraction of files from 20201217 to 20220128 with the stated difference between the hospital admissions reference date and the date in the filename. It's not uniform, even accounting for day of week. 

| Day of week (reference date) | count | delta=1 | delta=2 |
| - | - | - | - |
| Mon | 51  | 0.0980392  | 0.901961 |
| Tue | 55  | 0.0909091  | 0.909091 |
| Wed | 40  | 0.1  | 0.9 |
| Thu | 9  | 0.444444  | 0.555556 |
| Fri | 24  | 0.25  | 0.75 |
| Sat | 56  | 0.107143  | 0.892857 |
| Sun | 52  | 0.0961538  | 0.903846 |

Not sure what we want to do here. Maybe "R+2 (usually) or R+1 (10% chance)"

Reply:
Could I see your code (just to calculate the same for vaccinations)? I think I'd need to do it separately for the cumulative signals and the doses administered signals if I understand correctly.

Reply:
Sure thing:
```
# filename: demo.py
from delphi_dsew_community_profile.pull import *
from delphi_utils import read_params
import pandas as pd

for f in os.listdir("input_cache"):
    if not f.endswith(".xlsx"): continue
    publish_date = Dataset.parse_publish_date(f)
    workbook = pd.ExcelFile(os.path.join("input_cache",f))
    headers = pd.read_excel(
        workbook, sheet_name="States", header=None, nrows=1
    ).values.flatten().tolist()
    for h in headers:
        if Dataset.skip_overheader(h):
            continue
        dt = DatasetTimes.from_header(h, publish_date)
        if dt.hosp_reference_date is None:
            continue
        print(publish_date, dt.hosp_reference_date, dt.hosp_reference_date.strftime("%a"), (publish_date-dt.hosp_reference_date).days)
```

& then 
```
$ env/bin/python demo.py >demo.txt
$ awk '{print $3, $4}' demo.txt |sort |uniq -c |awk '{n+=$1; last=was; was=$1} NR%2==0 {print "| " $2, "| " n, " | " last/n, " | " was/n " |"; n=0}'
```

from an environment that has a populated `input_cache` directory. (sorry for the shell script, it was intended to be a one-off)

Reply:
For Cumulative Vaccinations: 

Day of week (reference date) | count | delta=1 | delta=2
-- | -- | -- | --
| Mon | 111  | 0.369369  | 0.630631 |
| Sun | 14  | 0.857143  | 0.142857 |
| Thu | 71  | 0.971831  | 0.028169 |
| Tue | 66  | 0.969697  | 0.030303 |
| Wed | 63  | 0.968254  | 0.031746 |

For Doses Administered: 

Day of week (reference date) | count | delta=1 | delta=2
-- | -- | -- | --
| Fri | 73  | 0.972603  | 0.0273973 |
| Fri | 4  | 0.5  | 0.5 |
| Mon | 98  | 0.0204082  | 0.979592 |
| Sun | 46  | 0.217391  | 0.782609 |
| Sun | 4  | 0.5  | 0.5 |
| Thu | 103  | 0.980583  | 0.0194175 |
| Tue | 92  | 0.978261  | 0.0217391 |
| Wed | 89  | 0.977528  | 0.0224719 |

So I'll change it to 1-2 days for vaccinations as well. 

Thanks Katie!